pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -634,154 +635,159 @@ def getData(self):
-                                bot_exc = (self._drlLastClip[1]-view_range.top()   ) / view_height
-                                # print(top_exc, bot_exc, hyst)
-                                if (    top_exc >= limit / hyst and top_exc <= limit * hyst
-                                    and bot_exc >= limit / hyst and bot_exc <= limit * hyst ):
-                                    # restore cached values
-                                    x = self.xDisp
-                                    y = self.yDisp
-                                    cache_is_good = True
-                            if not cache_is_good:
-                                min_val = view_range.bottom() - limit * view_height
-                                max_val = view_range.top()    + limit * view_height
-                                if( self.yDisp is not None              # Do we have an existing cache? 
-                                    and min_val >= self._drlLastClip[0] # Are we reducing it further?
-                                    and max_val <= self._drlLastClip[1] ):
-                                    # if we need to clip further, we can work in-place on the output buffer
-                                    # print('in-place:', end='')
-                                    # workaround for slowdown from numpy deprecation issues in 1.17 to 1.20+ :
-                                    # np.clip(self.yDisp, out=self.yDisp, a_min=min_val, a_max=max_val)
-                                    fn.clip_array(self.yDisp, min_val, max_val, out=self.yDisp)
-                                    self._drlLastClip = (min_val, max_val)
-                                    # print('{:.1e}<->{:.1e}'.format( min_val, max_val ))
-                                    x = self.xDisp
-                                    y = self.yDisp
-                                else:
-                                    # if none of the shortcuts worked, we need to recopy from the full data
-                                    # print('alloc:', end='')
-                                    # workaround for slowdown from numpy deprecation issues in 1.17 to 1.20+ :
-                                    # y = np.clip(y, a_min=min_val, a_max=max_val)
-                                    y = fn.clip_array(y, min_val, max_val)
-                                    self._drlLastClip = (min_val, max_val)
-                                    # print('{:.1e}<->{:.1e}'.format( min_val, max_val ))
-
-            self.xDisp = x
-            self.yDisp = y
-            self._viewRangeWasChanged = False
+        x = self.xData
+        y = self.yData
+        view = self.getViewBox()
+        if view is None:
+            v_rng = None
+        else:
+            v_rng = self.getViewBox().viewRect() # this is always up-to-date
+        if v_rng is None:
+            v_rng = self.viewRect()
+            # assert v_rng is None # halt if this actually did anything!
```

Comment:
use getViewBox().viewRect() unless it is not available.
Fall back to current behaviour of self.viewRect() (which does an inversion of the view transform to find the same numbers in all of my tests...) only if this is not available.

Reply:
Does this codepath need to be concerned with the situation when `self.getViewBox()` might return None?

Reply:
No, the previous code is just `range = self.viewRect()` .  A `None` result is handled by the individual clipping/limiting routines, mostly by simply not doing anything in that case. (I changed the name to v_rng to avoid shadowing the range command.)